media-service/src/main/java/org/arpha/service/MediaServiceImpl.java

```diff
@@ -0,0 +1,81 @@
+package org.arpha.service;
+
+import com.azure.storage.blob.BlobClient;
+import com.azure.storage.blob.BlobContainerClient;
+import com.querydsl.core.types.Predicate;
+import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
+import org.arpha.dto.media.request.FileUploadRequest;
+import org.arpha.dto.media.response.FileResponse;
+import org.arpha.entity.File;
+import org.arpha.exception.FileNotFoundException;
+import org.arpha.exception.FileUploadException;
+import org.arpha.mapper.FileMapper;
+import org.arpha.repository.FileRepository;
+import org.arpha.utils.Boxed;
+import org.arpha.validator.FileRequestValidator;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.util.UUID;
+
+import static org.arpha.dto.media.enums.AccessType.READ;
+import static org.arpha.dto.media.enums.AccessType.WRITE;
+
+@Service
+@RequiredArgsConstructor
+public class MediaServiceImpl implements MediaService {
+
+    private final BlobContainerClient blobContainerClient;
+    private final FileRequestValidator fileRequestValidator;
+    private final FileRepository fileRepository;
+    private final FileMapper fileMapper;
+
+    @Override
+    public FileResponse upload(FileUploadRequest fileUploadRequest) {
+        return Boxed
+                .of(UUID.randomUUID())
+                .mapToBoxed(UUID::toString)
+                .mapToBoxed(uuid-> uuid + "." + fileUploadRequest.getType().getSubtype())
```

Comment:
Consider adding targetType/targetId/ prefix to the file name. It will put the file itself into a virtual folder inside of azure blob storage

Reply:
Also what I do not like here - is that u r starting your Boxed chain with a UUID. Lets try to add more private methods that will be responsible for their own logic. For example, fileName generation can be encapsulated into generateFileName method. So, I suggest splitting a single Boxed chain to a smaller ones, each of which will be responsible for its own logic

Reply:
> Consider adding targetType/targetId/ prefix to the file name. It will put the file itself into a virtual folder inside of azure blob storage

What's the point of creating separated folder for users since he can have only one photo and in case he wants to change it should replaced or updated

Reply:
This endpoint will be used for all entities that require photos/videos etc. What if target type is product that can have multiple photos?

Reply:
> Quote reply
> Referen

About product I understand, but I asked about users

Reply:
> This endpoint will be used for all entities that require photos/videos etc. What if target type is product that can have multiple photos?

About product I understand, but I asked about users

Reply:
Well, users - u r right. But since we are creating generalized endpoint for uploading files for different purposes/targets, I still think that we should rely on virtual folders